extensions/amp-form/0.1/amp-form.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+import {getService} from '../../../src/service';
+import {xhrFor} from '../../../src/xhr';
+import {childElement, removeElement} from '../../../src/dom';
+import {templatesFor} from '../../../src/template';
+import {toArray} from '../../../src/types';
+import {onDocumentReady} from '../../../src/document-ready';
+import {isExperimentOn} from '../../../src/experiments';
+
+/** @const @type {string} */
+const TAG = 'amp-form';
+
+class AmpForm {
+
+  constructor(element, win) {
+    /** @const @private {!Window} */
+    this.win_ = win;
```

Comment:
you don't need to pass `win` here since you can get it from `element.ownerDocument.defaultView`